### PR TITLE
Add menu popup

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

File: kit/popups/src/Menu/UIMenuContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    );
+    const styles = useStyles(theme, menuLocation, shadowColor, shadowOpacity);
+
+    if (!visible) {
+        return null;
+    }
+
+    return (
+        <Portal absoluteFill>
+            <TouchableOpacity onPressIn={onPressIn} style={StyleSheet.absoluteFill} />
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainer.tsx]: Maybe make it consistent with alert? https://github.com/tonlabs/UIKit/blob/development/kit/popups/src/UIAlert/AlertBox.tsx#L142-L148

File: kit/popups/src/Menu/UIMenuContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        ColorVariants.BackgroundOverlay,
+    );
+    const styles = useStyles(theme, menuLocation, shadowColor, shadowOpacity);
+
+    if (!visible) {
+        return null;
+    }
+
+    return (
+        <Portal absoluteFill>
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainer.tsx]: Please introduce `forId` prop, as it's done for other popups, sometimes it's proved to be useful

File: kit/popups/src/Menu/UIMenuContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            <TouchableOpacity onPressIn={onPressIn} style={StyleSheet.absoluteFill} />
+            <Animated.View
+                style={styles.container}
+                entering={FadeIn.duration(UIConstant.menu.animationTime)}
+                exiting={FadeOut.duration(UIConstant.menu.animationTime)}
+                onLayout={onLayout}
+            >
+                <ShadowView style={styles.shadowContainer}>{children}</ShadowView>
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainer.tsx]: Have you tried to use `Animated.createAnimatedComponent` to remove wrapping `Animated.View`?

File: kit/popups/src/Menu/UIMenuContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        let bottom: number | undefined;
+        let left: number = targetDimensions.x;
+
+        const boundaries = getBoundaries(windowDimensions, menuSize);
+
+        if (left > boundaries.right) {
+            left = boundaries.right;
+        }
+        if (left < boundaries.left) {
+            left = boundaries.left;
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainer.tsx]: `Math.max(boundaries.left, Math.min(boundaries.right, targetDimensions.x))`

File: kit/popups/src/Menu/UIMenuContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        return null;
+    }
+
+    return (
+        <Portal absoluteFill>
+            <TouchableOpacity onPressIn={onPressIn} style={StyleSheet.absoluteFill} />
+            <Animated.View
+                style={styles.container}
+                entering={FadeIn.duration(UIConstant.menu.animationTime)}
+                exiting={FadeOut.duration(UIConstant.menu.animationTime)}
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainer.tsx]: cool! :)

File: kit/popups/src/Menu/UIMenuContainer.tsx (hunk 1 above)
Comment (savelichalex) [reply kit/popups/src/Menu/UIMenuContainer.tsx]: Also not sure that it should be `opacity` one...

File: kit/popups/src/Menu/UIMenuContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+            left,
+            top,
+            bottom,
+        };
+    }, [targetDimensions, windowDimensions, menuSize]);
+}
+
+export function UIMenuContainer({ children, visible, targetRef, onClose }: UIMenuContainerProps) {
+    const theme = useTheme();
+    const windowDimensions = useWindowDimensions();
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainer.tsx]: Frankly I don't like that all those computations are done outside of a `portal`, that means that if we had 100 menus, we would have 100 listeners hanging in memory.

What do you think to wrap it all with a `UIMenuContainerContent` component?

File: kit/popups/src/Menu/UIMenuContainer.tsx (hunk 3 above)
Comment (SAnatoliiS) [reply kit/popups/src/Menu/UIMenuContainer.tsx]: I tried it. There were problems with `ShadowView` performance, especially on android. It started lagging.

File: kit/popups/src/Menu/UIMenuContainer.tsx (hunk 6 above)
Comment (SAnatoliiS) [reply kit/popups/src/Menu/UIMenuContainer.tsx]: Great idea!

File: kit/popups/src/Menu/UIMenuContainer.tsx (hunk 3 above)
Comment (savelichalex) [reply kit/popups/src/Menu/UIMenuContainer.tsx]: Ok :)

**Review by savelichalex — APPROVED** [review]

File: kit/popups/package.json [hunk 7]
```diff
@@ -63,6 +64,7 @@
         "react-native-gesture-handler": "^1.10.3",
         "react-native-reanimated": "^2.3.0",
         "react-native-safe-area-context": "^3.3.2",
+        "react-native-simple-shadow-view": "1.6.3",
```
Comment (sertony) [inline kit/popups/package.json]:
```suggestion
        "react-native-simple-shadow-view": "^1.6.3",
```

File: kit/popups/src/Menu/UIMenuContainerContent.tsx [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+    return (
+        <>
+            <TapGestureHandler onEnded={onClose}>
+                <View style={StyleSheet.absoluteFill} />
+            </TapGestureHandler>
+            <Animated.View
+                style={styles.container}
+                entering={FadeIn.duration(UIConstant.menu.animationTime)}
+                exiting={FadeOut.duration(UIConstant.menu.animationTime)}
```
Comment (sertony) [inline kit/popups/src/Menu/UIMenuContainerContent.tsx:152]: 🚀

**Review by sertony — APPROVED** [review]

File: kit/popups/src/Menu/UIMenuContainerContent.tsx (hunk 8 above)
Comment (SAnatoliiS) [reply kit/popups/src/Menu/UIMenuContainerContent.tsx:152]: It will be improved in next PR 😎

**Pull request merged**